gensim/corpora/textcorpus.py

```diff
@@ -125,5 +211,121 @@ def __len__(self):
             # cache the corpus length
             self.length = sum(1 for _ in self.get_texts())
         return self.length
-
 # endclass TextCorpus
+
+
+class TextDirectoryCorpus(TextCorpus):
+    """Read documents recursively from a directory,
+    where each file is interpreted as a plain text document.
```

Comment:
I think we could add an option for that, but the corpora this was designed for (20 newsgroups, Wiki-Movie from coherence papers) are one document per file.